e2e/etcdctlv3_test.go

```diff
@@ -0,0 +1,144 @@
+// func TestCtlV3SetClientTLSQuorum(t *testing.T) {
+// 	checkTesting(t, testCtlV3Set(t, &configClientTLS, 5*time.Second, true))
+// }
+
+func TestCtlV3SetPeerTLSQuorum(t *testing.T) {
+	checkTesting(t, testCtlV3Set(t, &configPeerTLS, 5*time.Second, true))
+}
+
+// TODO: fix this. Getting transport: x509: certificate signed by unknown authority
+// func TestCtlV3SetTLSQuorum(t *testing.T) {
+// 	checkTesting(t, testCtlV3Set(t, &configTLS, 5*time.Second, true))
+// }
+
+func checkTesting(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func stripSchema(s string) string {
+	if strings.HasPrefix(s, "http://") {
+		s = strings.Replace(s, "http://", "", -1)
+	}
+	if strings.HasPrefix(s, "https://") {
+		s = strings.Replace(s, "https://", "", -1)
+	}
+	return s
+}
+
+func testCtlV3Set(t *testing.T, cfg *etcdProcessClusterConfig, dialTimeout time.Duration, quorum bool) error {
+	defer testutil.AfterTest(t)
+
+	epc := setupEtcdctlV3Test(t, cfg, quorum)
+	defer func() {
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	key, value := "foo", "bar"
+
+	if err := etcdctlV3Put(epc, key, value, dialTimeout); err != nil {
+		return fmt.Errorf("put error (%v)", err)
+	}
+
+	if err := etcdctlV3Get(epc, key, value, dialTimeout, quorum); err != nil {
+		return fmt.Errorf("get error (%v)", err)
+	}
+
+	return nil
+}
+
+func etcdctlV3PrefixArgs(clus *etcdProcessCluster, dialTimeout time.Duration) []string {
+	endpoints := ""
+	// TODO: add proxy check as in v2
+	if backends := clus.backends(); len(backends) != 0 {
+		es := []string{}
+		for _, b := range backends {
+			es = append(es, stripSchema(b.cfg.acurl.String()))
+		}
+		endpoints = strings.Join(es, ",")
+	}
+	cmdArgs := []string{"../bin/etcdctlv3", "--endpoints", endpoints, "--dial-timeout", dialTimeout.String()}
+	if clus.cfg.isClientTLS {
+		cmdArgs = append(cmdArgs, "--cacert", caPath, "--cert", certPath, "--key", privateKeyPath)
+	}
+	fmt.Println(cmdArgs)
+	return cmdArgs
+}
+
+func etcdctlV3Put(clus *etcdProcessCluster, key, value string, dialTimeout time.Duration) error {
+	cmdArgs := append(etcdctlV3PrefixArgs(clus, dialTimeout), "put", key, value)
+	return spawnWithExpectedString(cmdArgs, "OK")
+}
+
+func etcdctlV3Get(clus *etcdProcessCluster, key, value string, dialTimeout time.Duration, quorum bool) error {
+	// TODO: add serialized option
+	quorum = true // now linearlized read by default
+
+	cmdArgs := append(etcdctlV3PrefixArgs(clus, dialTimeout), "get", key)
+
+	// TODO: match by value. Currently it prints out both key and value in multi-lines
+	return spawnWithExpectedString(cmdArgs, key)
+}
+
+func mustEtcdctlV3(t *testing.T) {
+	if !fileutil.Exist("../bin/etcdctlv3") {
+		t.Fatalf("could not find etcdctlv3 binary")
+	}
+}
+
+func setupEtcdctlV3Test(t *testing.T, cfg *etcdProcessClusterConfig, quorum bool) *etcdProcessCluster {
+	// TODO: add serialized option
+	quorum = true // now linearlized read by default
```

Comment:
why override `quorum` here? it doesn't make sense for some commands (e.g., `defrag`)